### ISSUE TITLE
Looking For Core Team Members with Awesome Laravel/PHP Chops 

### ISSUE THREAD

**Issue by syropian** [astralapp/astral#31]
Hi! I'm **Syropian**, the creator of Astral.

I'm currently looking for one or more awesome Laravel developers who can contribute a few hours a week improving Astral's backend code. Though I know my way around Laravel for the most part, as a full-time front end developer, my strengths lie in that realm, and there are much more talented Laravel devs that could really make Astral's backend a lot better. 

Things that do/will need work on at the moment are:

- **Caching app-wide**: App-wide & intelligent caching could really speed up some of the responses from the server, and make the whole app feel snappier. I've made progress on v2 github star caching, but there's lots more to be done
- **Route/Controller Refactoring**: A rethinking of the routes and controllers, with the end goal of the backend being a nice, tidy API that will eventually even be able to offer 3rd party access to.
- **Tests!!!**: I'm notoriously bad about writing PHP tests. Help me Obi-Wan Kenobi. You're my only hope.

So if you love Astral (or maybe like what it could become with your help), and want to contribute to an awesome open source project, either leave a comment on this issue, or if you'd rather chat privately, shoot an email to hello@astralapp.com. 

As a core member you will receive attribution on Astral's website, and you can brag about it to all your friends too.  

![Neat](http://www.reactiongifs.com/wp-content/uploads/2013/07/neat.gif)

**Comment by sixlive** [comment]
I'd really like to say yes, I love the app, but it will be a few weeks for my schedule to clear up.

**Comment by syropian** [comment]
@sixlive No sweat dude, there is no hard deadline. Whenever you think you have the time, shoot me a message (or come say hey in the [Gitter channel](https://gitter.im/astralapp/astral)!).

**Comment by Williams94** [comment]
@syropian I know this was posted quite a while ago, but I'm not sure where else to contact you in regards to getting involved in this project. I have been using your app a lot in the past few months and find it very useful! Although there are a few problems I have run into, it is such a great app and getting a lot of attention even still today.

So I would love to offer my skills to help out as I am a Vue.js and Laravel Software Developer by day. I am very competent in both and I'm looking for an open source project to work on.

I can start with Route/Controller Refactoring & PHP tests if you're busy on v3 of the app. Just let me know what Laravel parts you need help with.

**Comment by Williams94** [comment]
Yes im definitely interested in helping out with v3! What means of communication do you want to use? If you let me know some of the parts you'd like me to work on,  I can get setup and start having a look. Then I'll be able to let you know how long I think they'll take me. Either frontend or backend stuff Im fine with, especially php unit testing if your still not keen on doing it yourself.

**Comment by syropian** [comment]
@Williams94 I could make a Slack or Discord server for us, whatever you prefer. I've got a _decent_ PHP test suite going on for v3, but there could probably be some improvements, and additional bits of coverage that could still be added. As for features, a couple things I have lined up are:

 - User settings
 - Multi-select (for tagging multiple stars with a single tag)
 - Migration page (there's a user-facing migration step that will need to happen when v3 launches)
 - Public star pages — a public page for people to share their stars

**Comment by m1guelpf** [comment]
Would love to help!

**Comment by Williams94** [comment]
@syropian Yeah either Slack or Discord would be good. They look like some good new features for v3! Yeah, I can set up some front-end testing as I've done it before and enjoyed it so shouldn't be a problem. My email is r.b.williams94@gmail.com so you can send me the details for your Slack or Discord channel when you get it set up. Looking forward to getting involved!

**Comment by syropian** [comment]
@Williams94 Great! Just shot you an email with a link to the discord.

**Comment by m1guelpf** [comment]
@syropian Maybe just post it here so I can also join? :smile:

**Comment by syropian** [comment]
@m1guelpf I don't want to open up the Discord publicly, but shoot an email to `collin@syropia.net` or DM me on Twitter `@syropian` and I'll send you an invite link.
